src/components/common/layout/header/Header.tsx

```diff
@@ -44,281 +18,25 @@ const Header: React.FC = () => {
-          </div>
-
-          <HeaderDivider />
-
-          {mobileMenu}
-        </div>
-      </div>
-    ) : (
-      <div className={styles['header-container']}>
-        <Link href={'/'}>
-          <div className={styles['header-logo']}>
-            <img src={`/assets/logo.png`} alt="logo" />
-          </div>
-        </Link>
-        <div className={styles['mobile-button']}>
-          <IconButton
-            onClick={handleClick}
-            size={IconButtonSize.MEDIUM}
-            color={IconButtonColor.TRANSPARENT}
-            icon={<BurgerMenu />}
-          />
-        </div>
-      </div>
-    );
-  }
-
-  return (
-    !isMobile && (
-      <div className={styles['header-container']}>
-        <Link href={'/'}>
-          <div className={styles['header-logo']}>
-            <img src={`/assets/logo.png`} alt="logo" />
-          </div>
-        </Link>
-        <div className={styles['menu']}>
-          <Link href={'/'}>
-            <Button
-              text="Головна"
-              size={ButtonSize.MEDIUM}
-              variant={ButtonVariant.TEXT}
-            />
-          </Link>
-          <Link href={'/poll'}>
-            <Button
-              text="Опитування"
-              size={ButtonSize.MEDIUM}
-              variant={ButtonVariant.TEXT}
-            />
-          </Link>
-          <Link href={'/teachers'}>
-            <Button
-              text="Викладачі"
-              size={ButtonSize.MEDIUM}
-              variant={ButtonVariant.TEXT}
-            />
-          </Link>
-          <Link href={'/subjects'}>
-            <Button
-              text="Предмети"
-              size={ButtonSize.MEDIUM}
-              variant={ButtonVariant.TEXT}
-            />
-          </Link>
-
-          {/* <Link href={{}}>
-            <Button
-              text="Розклад"
-              size={ButtonSize.MEDIUM}
-              variant={ButtonVariant.TEXT}
-            />
-          </Link> */}
-        </div>
-        {isLoggedIn ? (
-          <div className={styles['header-desktop-card']}>
-            <Link href={'/account'}>
-              <HeaderDesktopCard
-                name={name}
-                groupName={groupName}
-                position={position}
-                url={avatar}
-              />
-            </Link>
-          </div>
-        ) : (
-          <div className={styles['login-buttons']}>
-            <Link href={'/register'}>
-              <Button
-                text="Зареєструватись"
-                size={ButtonSize.SMALL}
-                variant={ButtonVariant.OUTLINE}
-              />
-            </Link>
-            <Link href={'/login'}>
-              <Button
-                text="Увійти"
-                size={ButtonSize.SMALL}
-                variant={ButtonVariant.FILLED}
-              />
-            </Link>
-          </div>
-        )}
-      </div>
-    )
+  const isMobile = useMediaQuery('(max-width:1200px)');
+  return isMobile ? (
+    <MobileHeader
```

Comment:
Let MobileHeader and DesktopHeader have 2 props:
1. user (object, which transformData will return)
2. isLoggedIn